src/test/java/space/obminyashka/items_exchange/controller/UserControllerTest.java

```diff
@@ -78,116 +81,57 @@ void negativeTestReceivingInformationAboutAnotherUser() throws Exception {
     }
 
     @Test
-    @WithMockUser(username = "admin")
-    void updateUserInfo_shouldReturn403WhenUsernameIsChanged() throws Exception {
-        when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
-        when(userService.update(any(), any())).thenThrow(new IllegalOperationException(
-                getMessageSource("exception.illegal.field.change") + user.getUsername()));
-
-        MvcResult result = getResultActions(HttpMethod.PUT, "/user/info",
-                createUserDtoForUpdatingWithChangedUsernameWithoutPhones(), status().isForbidden())
-                .andDo(print())
-                .andReturn();
-        assertTrue(result.getResponse().getContentAsString().contains(
-                getMessageSource("exception.illegal.field.change") + user.getUsername()));
-    }
-
-    @Test
-    @WithMockUser(username = "admin")
-    void updateUserInfo_shouldReturn403WhenLastOnlineTimeIsChanged() throws Exception {
+    @WithMockUser(username = "deletedUser")
+    void updateUserInfo_WhenUserHasStatusDeleted_ShouldReturn403WithSpecificMessage() throws Exception {
+        user.setStatus(Status.DELETED);
         when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
-        when(userService.update(any(), any())).thenThrow(new IllegalOperationException(
-                getMessageSource("exception.illegal.field.change") + user.getLastOnlineTime()));
+        when(userService.getDaysBeforeDeletion(any())).thenReturn(7L);
 
-        MvcResult result = getResultActions(HttpMethod.PUT, "/user/info",
-                createUserDtoForUpdatingWithChangedLastOnlineTimeWithoutPhones(), status().isForbidden())
+        MvcResult mvcResult = getResultActions(HttpMethod.PUT, "/user/info",
+                createUserDtoForUpdatingWithChangedEmailAndFNameApAndLNameMinusWithoutPhones(),
+                status().isForbidden())
                 .andDo(print())
                 .andReturn();
-        assertTrue(result.getResponse().getContentAsString().contains(
-                getMessageSource("exception.illegal.field.change") + user.getLastOnlineTime()));
-    }
 
-    @Test
-    @WithMockUser(username = "admin")
-    void updateUserInfo_shouldReturn403WhenChildrenAreChanged() throws Exception {
-        when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
-        when(userService.update(any(), any())).thenThrow(new IllegalOperationException(
-                getMessageSource("exception.illegal.field.change") + user.getChildren()));
+        User deletedUser = userService.findByUsernameOrEmail("deletedUser@gmail.com").orElseThrow();
```

Comment:
U already have the user which was used for mock data, and now you get another one. Could you explain why?

Reply:
it isn`t my logic, however thanks. I revised this method again in more detail and made some changes